src/hooks/privateChats.ts

```diff
@@ -215,3 +225,61 @@ export const useRecipientChat = (recipientId: string) => {
     recipient,
   };
 };
+
+export const usePrevious = <T>(value: T) => {
+  const ref = useRef<T>(value);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+export const usePrivateChatNotification = (
+  privateChatMessages: WithId<PrivateChatMessage>[]
+) => {
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(newMessageSound, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const privateChateMessagesSorted = useMemo(
+    () => Object.values(privateChatMessages).sort(chatSort),
+    [privateChatMessages]
+  );
+
+  const prevPrivateChatMessages = usePrevious(privateChateMessagesSorted[0]);
+
+  const currentChatUserId =
+    chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT
+      ? chatSettings.recipientId
+      : undefined;
+
+  return useMemo(() => {
+    const shouldPlaySound = privateChateMessagesSorted
+      .filter((message) => {
+        return (
+          message?.ts_utc.seconds > prevPrivateChatMessages?.ts_utc.seconds
+        );
+      })
+      .some(
+        (message) =>
+          !message.isRead && ![userId, currentChatUserId].includes(message.from)
+      );
+
+    if (shouldPlaySound) {
+      play();
+    }
+  }, [
+    privateChateMessagesSorted,
+    currentChatUserId,
+    play,
+    prevPrivateChatMessages,
+    userId,
+  ]);
```

Comment:
Why is this using `useMemo` here? And what are we returning (I don't think anything)?

In my last review comments describing this solution, I talked about using `useEffect` here? (https://github.com/sparkletown/sparkle/pull/1262#discussion_r604633362) Why didn't we do that?

I expect that's what we should be doing here + probably removing the `return`, as I don't think we need it do we?